conf/broker.conf

```diff
@@ -517,6 +517,13 @@ exposePublisherStats=true
 # The schema storage implementation used by this broker
 schemaRegistryStorageClassName=org.apache.pulsar.broker.service.schema.BookkeeperSchemaStorageFactory
 
+# Enforce schema validation on following cases:
+#
+# - if a producer without a schema attempts to produce to a topic with schema, the producer will be
+#   failed to connect. PLEASE be carefully on using this, since non-java clients don't support schema.
+#   if you enable this setting, it will cause non-java clients failed to produce.
+isSchemaValidationEnforced=false
```

Comment:
I think enforcing schema validation should not be at broker level and such validation should be configured at namespace level. So, I think default schema should be optional at broker and we should have namespace-admin api so, any tenant can enable validation at namespace-policies.??

Reply:
yes agreed, such api should be controlled at namespace level. the change here is more providing a flag for smooth upgrading story. the flag is more for the people who operates the whole pulsar cluster, and the namespace level is for tenants. to me, namespace level control is a feature to provide fine granualarity on management, the flag here is make sure the shema enforcement can be enabled in a BC way when people upgrade a pulsar cluster.

does that make sense to you?

Reply:
but with this option, isn't a chance that producer without schema can inject incompatible data. So, it's better to reject the request without schema, if schema is enabled for that topic.? 
however, we can always remove this flag in future if we don't see any meaning.

Reply:
> So, it's better to reject the request without schema, if schema is enabled for that topic.?
> however, we can always remove this flag in future if we don't see any meaning.

yes, we should remove this flag after 2.3 or 2.4. before getting there, we also need to get a upgrade guide ready :)

Reply:
👍 For 2.2 this flag is already enough. Later we should add namespace level control